sstables/sstables_manager.cc

```diff
@@ -19,8 +19,13 @@ namespace sstables {
 logging::logger smlogger("sstables_manager");
 
 sstables_manager::sstables_manager(
-    db::large_data_handler& large_data_handler, const db::config& dbcfg, gms::feature_service& feat, cache_tracker& ct)
-    : _large_data_handler(large_data_handler), _db_config(dbcfg), _features(feat), _cache_tracker(ct) {
+    db::large_data_handler& large_data_handler, const db::config& dbcfg, gms::feature_service& feat, cache_tracker& ct, size_t available_memory)
+    : _large_data_handler(large_data_handler), _db_config(dbcfg), _features(feat), _cache_tracker(ct)
+    , _sstable_metadata_concurrency_sem(
+        max_count_sstable_metadata_concurrent_reads,
+        max_memory_sstable_metadata_concurrent_reads(available_memory),
```

Comment:
If you limit to all available memory, you'll still get OOMs. Need to limit to a subset.

Reply:
Oh, it's a function call. So ignore.

Reply:
Yes. Will add comment explaining the choice, as suggested by Benny.